packages/services/src/contents/index.ts

```diff
@@ -477,6 +495,12 @@ export namespace Contents {
    * in the contents manager.
    */
   export interface IDrive extends IDisposable {
+    /**
+     * The content provider registry, consisting of all the
+     * content providers that this drive can use to access files.
+     */
+    readonly contentProviderRegistry: IContentProviderRegistry;
```

Comment:
Technically this adds a new required attribute, so has a potential to break existing extensions. Can we make it optional for now?

```suggestion
    readonly contentProviderRegistry?: IContentProviderRegistry;
```

That said, the risk is low because most extensions would just inherit from `Drive`.